File: protocolAggregator/contracts/src/main/java/com/protocol180/aggregator/states/ProviderAggregationState.java
package com.protocol180.aggregator.states;

import com.google.common.collect.ImmutableList;
import net.corda.core.contracts.ContractState;
import net.corda.core.contracts.LinearState;
import net.corda.core.contracts.StaticPointer;
import net.corda.core.contracts.UniqueIdentifier;
import net.corda.core.identity.AbstractParty;
import net.corda.core.identity.AnonymousParty;
import net.corda.core.identity.Party;
import org.jetbrains.annotations.NotNull;

import java.util.List;

public class ProviderAggregationState implements ContractState {
    public final AnonymousParty provider;
    public final Party host;
    public final String failedReason;
    public final byte[] dataInput;
    private final StaticPointer<ConsumerAggregationState> pointedTolAggregationState;
    private final StaticPointer<RewardsState> pointedToRewardState;


    public ProviderAggregationState(AnonymousParty provider, Party host, String failedReason, byte[] dataInput, StaticPointer<ConsumerAggregationState> pointedTolAggregationState, StaticPointer<RewardsState> pointedToRewardState) {
        this.provider = provider;
        this.host = host;
        this.failedReason = failedReason;
        this.dataInput = dataInput;
        this.pointedTolAggregationState = pointedTolAggregationState;
        this.pointedToRewardState = pointedToRewardState;
    }

    @NotNull
    @Override
    public List<AbstractParty> getParticipants() {
        return ImmutableList.of(provider, host);
    }

    public String getFailedReason() {
        return failedReason;
    }

    public byte[] getDataInput() {
        return dataInput;
    }

    public StaticPointer<ConsumerAggregationState> getPointedTolAggregationState() {
        return pointedTolAggregationState;
    }

    public StaticPointer<RewardsState> getPointedToRewardState() {
        return pointedToRewardState;
    }
}

File: protocolAggregator/contracts/src/main/kotlin/com/protocol180/aggregator/states/ProviderAggregationState.kt
package com.protocol180.aggregator.states

import com.protocol180.aggregator.contracts.AggregationContract
import com.protocol180.commons.SchemaType
import net.corda.core.contracts.BelongsToContract
import net.corda.core.contracts.ContractState
import net.corda.core.contracts.StaticPointer
import net.corda.core.identity.AbstractParty
import net.corda.core.identity.AnonymousParty
import net.corda.core.identity.Party

@BelongsToContract(AggregationContract::class)
data class ProviderAggregationState(val provider: AnonymousParty,
                                    val host: Party,
                                    val failedReason:String,
                                    val dataInput: ByteArray,
                                    val pointedToConsumerAggregationState: StaticPointer<ConsumerAggregationState>,
                                    val pointedToRewardsState: StaticPointer<RewardsState>) : ContractState {

    /**
     *  This property holds a list of the nodes which can "use" this state in a valid transaction. In this case, the
     *  provider or host.
     */
    override val participants: List<AbstractParty> get() = listOf(provider, host)

}